osu.Game/Overlays/IDialogOverlay.cs

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+#nullable enable
+
+using osu.Framework.Allocation;
+using osu.Game.Overlays.Dialog;
+
+namespace osu.Game.Overlays
+{
+    /// <summary>
+    /// A global overlay that can show popup dialogs.
+    /// </summary>
+    [Cached]
```

Comment:
For correctness, this should be cached as type `IDialogOverlay`.
```suggestion
    [Cached(typeof(IDialogOverlay))]
```

Though I'm not sure if doing this will be of any benefit, since it only means the dialog overlay will be cached to its children (`PopupDialog`/`PopupDialogButton`/etc.)